packages/react-composites/src/composites/ChatComposite/adapter/ChatAdapter.ts

```diff
@@ -15,6 +17,20 @@ export type ChatAdapterUiState = {
   // FIXME(Delete?)
   // Self-contained state for composite
   error?: Error;
+  /* @conditional-compile-remove-from(stable): FILE_SHARING */
+  /**
+   * Files being uploaded by a user in the current thread.
+   * Should be set to null once the upload is complete.
+   * @beta
+   */
+  uploadedFiles?: UploadedFile[];
```

Comment:
In-line with my comment about the interface used by the file upload callback, this should not store the full `UploadedFile` object. It should _only_ contain the information that is needed in the adapter.

My assumption (right now) is that what is needed in the adapter state is:
* For each active file transfer, the name, extension (for icon), progress of the file transfer (i.e., read-only data).

I'd recommend that we create a new type that only contains those fields.

It can't directly use `UploadedFile` anyway, because all adapter state must be immutable (but `UploadedFile` is a mutable class).